client/van_router_create_test.go

```diff
@@ -193,10 +208,21 @@ func TestVanRouterCreateDefaults(t *testing.T) {
 		svcAccountsFound := []string{}
 
 		// Create the client
-		cli, err := newMockClient(c.namespace, "", "")
+		var cli *VanClient
+		var err error
+		if !isCluster {
+			cli, err = newMockClient(c.namespace, "", "")
+		} else {
+			cli, err = NewClient(c.namespace, "", "")
+		}
 		assert.Check(t, err, c.doc)
 
-		informers := informers.NewSharedInformerFactory(cli.KubeClient, 0)
+		if isCluster {
+			_, err = kube.NewNamespace(c.namespace, cli.KubeClient)
+			assert.Check(t, err, c.doc)
```

Comment:
maybe the "delete namespace" can be deferred here, so you are sure it is removed, no matter what kind of error you have in the test